tests/data/test_HOA_tools_cec2.py

```diff
@@ -1,146 +1,251 @@
-        (1, 1, 0.8840922122960315),
-        (45, 54, 0.2378095805520779),
-        (18, 91, 0.39897595320190293),
+        (1234, 34.9827354617758),
+        (1235, 37.63083581519224),
+        (1236, 35.83739821236641),
+    ],
+)
+def test_hoa_rotate(seed: int, expected: float) -> None:
+    """Test the HOARotator rotate method"""
+
+    # 1st order ambinsonic signal, ie 4 channels
+    np.random.seed(seed)
+    signal = np.random.random((20, 4))
+    rotations = np.random.random(20)
+    hoa_rotator = HOARotator(1, 90)
+    result = hoa_rotator.rotate(signal, rotations)
+    assert np.sum(np.abs(result)) == pytest.approx(expected)
+
+
+@pytest.mark.parametrize(
+    "seed, row, col, expected",
+    [
+        (1234, 1, 1, 0.740996817336437),
+        (1235, 45, 24, 0.2617612929455504),
+        (1236, 18, 47, 0.2706397841499083),
     ],
 )
 def test_centred_element(
-    random_matrix: np.ndarray, row: int, col: int, expected: float
+    make_random_matrix, seed: int, row: int, col: int, expected: float
 ) -> None:
     """Test for centered_element() function."""
-    assert centred_element(random_matrix, row, col) == pytest.approx(
-        expected, rel=pytest.rel_tolerance, abs=pytest.abs_tolerance
-    )
+    random_matrix = make_random_matrix(seed)
+    assert centred_element(random_matrix, row, col) == pytest.approx(expected)
 
 
-def test_centred_element_index_error(random_matrix: np.ndarray) -> None:
+def test_centred_element_index_error(make_random_matrix) -> None:
     """Test that centered_element() raises an IndexError if centering is outside
     of matrix dimensions."""
+    random_matrix = make_random_matrix()
     with pytest.raises(IndexError):
         centred_element(random_matrix, row=101, col=2)
     with pytest.raises(IndexError):
         centred_element(random_matrix, row=1, col=102)
 
 
 @pytest.mark.parametrize(
-    "i, a, b, order, expected",
+    "i, a, b, order, seed, expected",
     [
-        (1, 1, 2, 2, -0.3475073926260553),
-        (1, 1, 2, -2, 0.6814353682034401),
-        (1, 1, 2, 2, -0.7185822065660934),
+        (1, 1, 2, 2, 1234, -0.20056163713681976),
+        (1, 1, 2, -2, 1235, 0.1767992365183524),
+        (1, 1, 2, 2, 1236, -0.5552636713315029),
     ],
 )
 def test_P(
-    i: int, a: int, b: int, order: int, random_matrix: np.ndarray, expected: float
+    i: int, a: int, b: int, order: int, make_random_matrix, seed: int, expected: float
 ) -> None:
     """Test for P() function."""
+    r1_matrix = make_random_matrix(seed)
+    r2_matrix = make_random_matrix(seed + 10)  # Different seed to get different matrix
+    r3_matrix = make_random_matrix(seed + 20)  # ditto
     assert P(
         i,
         a,
         b,
         order,
-        rotation_matrices=[random_matrix, random_matrix, random_matrix],
-    ) == pytest.approx(expected, rel=pytest.rel_tolerance, abs=pytest.abs_tolerance)
+        rotation_matrices=TypedList([r1_matrix, r2_matrix, r3_matrix]),
+    ) == pytest.approx(expected)
```

Comment:
Is the default relative tolerance of `1e-6` ok for this test or broadly across the whole code base? 

If so we could change those defaults in `conftest.default__config()` so they are used globally and always explicitly state the values should we ever want to change them across the code base.